current/runtime/storage.md

```diff
@@ -4,49 +4,240 @@ lang: en
+algorithm would produce a wide distribution of outputs even if the inputs were the numbers 1
+through 10. It is critical to use cryptographic hashing algorithms when users are able to influence
+the keys of a Storage Map. Failure to do so creates an attack vector that makes it easy for
+malicious actors to degrade the performance of your blockchain network. An example of a map that
+should use a cryptographic hash algorithm to generate its keys is a map used to track account
+balances. In this case, it is important to use a cryptographic hashing algorithm so that an attacker
+cannot bombard your system with many small transfers to sequential account numbers; without a
+cryptographic hash algorithm this would create an imbalanced storage structure that would suffer in
+performance. Cryptographic hashing algorithms are more complex and resource-intensive than their
+non-cryptographic counterparts, which is why Substrate allows developers to select when they are
+used.
+
+##### Transparent Hashing Algorithms
+
+A transparent hashing algorithm is one that makes it easy to discover and verify the input that was
+used to generate a given output. In Substrate, hashing algorithms are made transparent by
+concatenating the algorithm's input to its output. This makes it trivial for users to retrieve a
+key's original unhashed value and verify it if they'd like (by re-hashing it). The creators of
+Substrate have **deprecated the use of non-transparent hashers** within FRAME-based runtimes, so
+this information is provided primarily for completeness. In fact, it is _necessary_ to use a
+transparent hashing algorithm if you would like access [iterable map](#Iterable-Storage-Maps)
+capabilities. Refer to [the advanced storage documentation](../advanced/storage#Storage-Map-Keys) to
+learn more about the important capabilities that transparent hashing algorithms expose.
+
+##### Common Substrate Hashers
+
+This table lists some common hashers used in Substrate and denotes those that are cryptographic and
+those that are transparent:
+
+| Hasher                                                                                     | Cryptographic | Transparent |
+| ------------------------------------------------------------------------------------------ | ------------- | ----------- |
+| [Blake2 128 Concat](https://crates.parity.io/frame_support/struct.Blake2_128Concat.html)   | X             | X           |
+| [TwoX 64 Concat](https://crates.parity.io/frame_support/struct.Twox64Concat.html)          |               | X           |
+| [Identity](https://crates.parity.io/frame_support/struct.Identity.html)                    |               |             |
+| [Blake2 128](https://crates.parity.io/frame_support/struct.Blake2_128.html) **DEPRECATED** | X             |             |
+| [TwoX 128](https://crates.parity.io/frame_support/struct.Twox128.html) **DEPRECATED**      |               |             |
```

Comment:
there is actually 4 depracated hasher but I don't mind if we don't mention them:



 >   twox_128 - TwoX with 128bit.
    twox_256 - TwoX with with 256bit.
    blake2_128 - Blake2 with 128bit.
    blake2_256 - Blake2 with 256bit.